Report should print issue title

Right now an issue is displayed like that:

```
hadrian/src/Settings/Packages.hs:205:30: error:
  now Closed: https://github.com/snowleopard/hadrian/issues/90
```

It can be really interesting if it was displayed with the issue title:

```
hadrian/src/Settings/Packages.hs:205:30: error:
  now Closed: https://github.com/snowleopard/hadrian/issues/90
  title: Cannot Build: compareSizes (rts missing _TRACE_gc and _TRACE_sched)
```

--- a/krank.cabal
+++ b/krank.cabal
@@ -66,6 +66,7 @@ test-suite krank-test
                        Utils.Display
                        Utils.Github
                        Utils.Gitlab
+                       Utils.Req
   build-depends: hspec >= 2.7
                , hspec-expectations
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N

--- a/src/Krank/Checkers/IssueTracker.hs
+++ b/src/Krank/Checkers/IssueTracker.hs
@@ -8,7 +8,7 @@
 {-# LANGUAGE ViewPatterns #-}
 
 module Krank.Checkers.IssueTracker
-  ( GitIssue (..),
+  ( GitIssueRef (..),
     GitServer (..),
     Localized (..),
     checkText,
@@ -39,8 +39,8 @@ data GitServer = Github | Gitlab GitlabHost
 
 data IssueStatus = Open | Closed deriving (Eq, Show)
 
-data GitIssue
-  = GitIssue
+data GitIssueRef
+  = GitIssueRef
       { server :: GitServer,
         owner :: Text,
         repo :: Text,
@@ -50,7 +50,7 @@ data GitIssue
 
 data GitIssueWithStatus
   = GitIssueWithStatus
-      { gitIssue :: Localized GitIssue,
+      { gitIssue :: Localized GitIssueRef,
         issueStatus :: IssueStatus
       }
   deriving (Eq, Show)
@@ -68,10 +68,10 @@ gitRepoRe :: RE.Regex
 gitRepoRe = [RE.re|\b(?>https?://)?(?>www\.)?([^/ ]+)/([^ ]+)/([^- ][^/ ]*)(?>/-)?/issues/([0-9]+)|]
 
 -- | Extract all issues on one line and returns a list of the raw text associated with an issue
-extractIssuesOnALine :: ByteString -> [(Int, GitIssue)]
+extractIssuesOnALine :: ByteString -> [(Int, GitIssueRef)]
 extractIssuesOnALine lineContent = map f (RE.scan gitRepoRe lineContent)
   where
-    f (match, [domain, owner, repo, ByteString.readInt -> Just (issueNo, _)]) = (colNo, GitIssue provider (Text.Encoding.decodeUtf8 owner) (Text.Encoding.decodeUtf8 repo) issueNo)
+    f (match, [domain, owner, repo, ByteString.readInt -> Just (issueNo, _)]) = (colNo, GitIssueRef provider (Text.Encoding.decodeUtf8 owner) (Text.Encoding.decodeUtf8 repo) issueNo)
       where
         colNo = 1 + ByteString.length (fst $ ByteString.breakSubstring match lineContent)
         provider
@@ -97,7 +97,7 @@ extractIssues ::
   FilePath ->
   -- | Content of the file
   ByteString ->
-  [Localized GitIssue]
+  [Localized GitIssueRef]
 extractIssues filePath toCheck = concat (zipWith extract [1 ..] (ByteString.lines toCheck))
   where
     extract lineNo lineContent = map f (extractIssuesOnALine lineContent)
@@ -106,7 +106,7 @@ extractIssues filePath toCheck = concat (zipWith extract [1 ..] (ByteString.line
 
 -- Supports only github for the moment
 issueUrl ::
-  GitIssue ->
+  GitIssueRef ->
   Req.Url 'Req.Https
 issueUrl issue = case server issue of
   Github -> Req.https "api.github.com" Req./: "repos" Req./: owner issue Req./: repo issue Req./: "issues" Req./~ issueNum issue
@@ -115,7 +115,7 @@ issueUrl issue = case server issue of
 -- try Issue can fail, on non-2xx HTTP response
 tryRestIssue ::
   MonadKrank m =>
-  Localized GitIssue ->
+  Localized GitIssueRef ->
   m Value
 tryRestIssue locIssue = do
   let issue = unLocalized locIssue
@@ -125,7 +125,7 @@ tryRestIssue locIssue = do
 
 headersFor ::
   MonadKrank m =>
-  GitIssue ->
+  GitIssueRef ->
   m (Req.Option 'Req.Https)
 headersFor issue = do
   mGithubKey <- krankAsks githubKey
@@ -155,7 +155,7 @@ showGitServerException (Gitlab _) exc = showGitlabException exc
 
 restIssue ::
   MonadKrank m =>
-  Localized GitIssue ->
+  Localized GitIssueRef ->
   m Value
 restIssue issue = catch (tryRestIssue issue) (httpExcHandler . server . unLocalized $ issue)
 
@@ -186,8 +186,8 @@ errorParser o = do
 
 gitIssuesWithStatus ::
   MonadKrank m =>
-  [Localized GitIssue] ->
-  m [Either (Text, Localized GitIssue) GitIssueWithStatus]
+  [Localized GitIssueRef] ->
+  m [Either (Text, Localized GitIssueRef) GitIssueWithStatus]
 gitIssuesWithStatus issues = do
   statuses <- krankMapConcurrently restIssue issues
   pure $ zipWith f issues (fmap statusParser statuses)
@@ -209,8 +209,8 @@ issueToMessage i = case issueStatus i of
   Open -> [fmt|still Open|]
   Closed -> [fmt|now Closed - You can remove the workaround you used there|]
 
-issuePrintUrl :: GitIssue -> Text
-issuePrintUrl GitIssue {owner, repo, server, issueNum} = [fmt|IssueTracker check for https://{serverDomain server}/{owner}/{repo}/issues/{issueNum}|]
+issuePrintUrl :: GitIssueRef -> Text
+issuePrintUrl GitIssueRef {owner, repo, server, issueNum} = [fmt|IssueTracker check for https://{serverDomain server}/{owner}/{repo}/issues/{issueNum}|]
 
 checkText ::
   MonadKrank m =>
@@ -229,7 +229,7 @@ checkText path t = do
                 { checker = issuePrintUrl . unLocalized $ issue,
                   level = Info,
                   message = "Dry run",
-                  location = getLocation (issue :: Localized GitIssue)
+                  location = getLocation (issue :: Localized GitIssueRef)
                 }
           )
           issues
@@ -242,12 +242,12 @@ checkText path t = do
         { checker = issuePrintUrl . unLocalized $ issue,
           level = Warning,
           message = "Error when calling the API:\n" <> err,
-          location = getLocation (issue :: Localized GitIssue)
+          location = getLocation (issue :: Localized GitIssueRef)
         }
     f (Right issue) =
       Violation
         { checker = issuePrintUrl (unLocalized . gitIssue $ issue),
           level = issueToLevel issue,
           message = issueToMessage issue,
-          location = getLocation (gitIssue issue :: Localized GitIssue)
+          location = getLocation (gitIssue issue :: Localized GitIssueRef)
         }

--- a/src/Krank/Checkers/IssueTracker.hs
+++ b/src/Krank/Checkers/IssueTracker.hs
@@ -51,7 +51,8 @@ data GitIssueRef
 data GitIssueData
   = GitIssueData
       { gitIssue :: Localized GitIssueRef,
-        issueStatus :: IssueStatus
+        issueStatus :: IssueStatus,
+        issueTitle :: Text
       }
   deriving (Eq, Show)
 
@@ -174,6 +175,17 @@ statusParser (AesonT.Object o) = do
     readState (AesonT.Error _) = Left $ errorParser o
 statusParser _ = Left "invalid JSON"
 
+titleParser ::
+  Value ->
+  Either Text Text
+titleParser (AesonT.Object o) = do
+  let title :: AesonT.Result String = AesonT.parse (.: "title") o
+  Right $ readTitle title
+  where
+    readTitle (AesonT.Success title) = pack title
+    readTitle (AesonT.Error _) = "invalid JSON"
+titleParser _ = Left "invalid JSON"
+
 errorParser ::
   AesonT.Object ->
   Text
@@ -189,11 +201,14 @@ gitIssuesWithStatus ::
   [Localized GitIssueRef] ->
   m [Either (Text, Localized GitIssueRef) GitIssueData]
 gitIssuesWithStatus issues = do
-  statuses <- krankMapConcurrently restIssue issues
-  pure $ zipWith f issues (fmap statusParser statuses)
+  jsonData <- krankMapConcurrently restIssue issues
+  let statuses = fmap statusParser jsonData
+  let titles = fmap titleParser jsonData
+  pure $ zipWith3 f issues statuses titles
   where
-    f issue (Left err) = Left (err, issue)
-    f issue (Right is) = Right $ GitIssueData issue is
+    f issue (Left err) _ = Left (err, issue)
+    f issue _ (Left err) = Left (err, issue)
+    f issue (Right status) (Right title) = Right $ GitIssueData issue status title
 
 issueToLevel ::
   GitIssueData ->

--- a/src/Krank/Checkers/IssueTracker.hs
+++ b/src/Krank/Checkers/IssueTracker.hs
@@ -220,9 +220,12 @@ issueToLevel i = case issueStatus i of
 issueToMessage ::
   GitIssueData ->
   Text
-issueToMessage i = case issueStatus i of
-  Open -> [fmt|still Open|]
-  Closed -> [fmt|now Closed - You can remove the workaround you used there|]
+issueToMessage i =
+  case issueStatus i of
+    Open -> [fmt|the issue is still Open\ntitle: {title}|]
+    Closed -> [fmt|the issue is now Closed - You can remove the workaround you used there\ntitle: {title}|]
+  where
+    title = issueTitle i
 
 issuePrintUrl :: GitIssueRef -> Text
 issuePrintUrl GitIssueRef {owner, repo, server, issueNum} = [fmt|IssueTracker check for https://{serverDomain server}/{owner}/{repo}/issues/{issueNum}|]

--- a/src/Krank/Checkers/IssueTracker.hs
+++ b/src/Krank/Checkers/IssueTracker.hs
@@ -48,8 +48,8 @@ data GitIssueRef
       }
   deriving (Eq, Show)
 
-data GitIssueWithStatus
-  = GitIssueWithStatus
+data GitIssueData
+  = GitIssueData
       { gitIssue :: Localized GitIssueRef,
         issueStatus :: IssueStatus
       }
@@ -187,23 +187,23 @@ errorParser o = do
 gitIssuesWithStatus ::
   MonadKrank m =>
   [Localized GitIssueRef] ->
-  m [Either (Text, Localized GitIssueRef) GitIssueWithStatus]
+  m [Either (Text, Localized GitIssueRef) GitIssueData]
 gitIssuesWithStatus issues = do
   statuses <- krankMapConcurrently restIssue issues
   pure $ zipWith f issues (fmap statusParser statuses)
   where
     f issue (Left err) = Left (err, issue)
-    f issue (Right is) = Right $ GitIssueWithStatus issue is
+    f issue (Right is) = Right $ GitIssueData issue is
 
 issueToLevel ::
-  GitIssueWithStatus ->
+  GitIssueData ->
   ViolationLevel
 issueToLevel i = case issueStatus i of
   Open -> Info
   Closed -> Error
 
 issueToMessage ::
-  GitIssueWithStatus ->
+  GitIssueData ->
   Text
 issueToMessage i = case issueStatus i of
   Open -> [fmt|still Open|]

--- a/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
+++ b/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
@@ -65,7 +65,7 @@ instance MonadKrank TestKrank where
           Data.Aeson.Error s -> throw (Req.JsonHttpException s)
         Left exception -> throw exception
 
-check :: ByteString -> Maybe GitIssue
+check :: ByteString -> Maybe GitIssueRef
 check a = case extractIssuesOnALine a of
   [(_, x)] -> Just x
   _ -> Nothing
@@ -75,19 +75,19 @@ giturlTests domain = do
   let domainName = serverDomain domain
   it "handles full https url" $ do
     let match = check [fmt|https://{domainName}/guibou/krank/issues/2|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 2)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 2)
   it "handles full http url" $ do
     let match = check [fmt|http://{domainName}/guibou/krank/issues/1|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 1)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 1)
   it "handles short url - no protocol" $ do
     let match = check [fmt|{domainName}/guibou/krank/issues/1|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 1)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 1)
   it "accepts www. in url" $ do
     let match = check [fmt|https://www.{domainName}/guibou/krank/issues/1|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 1)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 1)
   it "accepts www in url - no protocol" $ do
     let match = check [fmt|www.{domainName}/guibou/krank/issues/1|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 1)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 1)
   it "fails if the issue number is not an int" $ do
     let match = check [fmt|{domainName}/guibou/krank/issues/foo|]
     match `shouldBe` Nothing
@@ -99,13 +99,13 @@ giturlTests domain = do
     match `shouldBe` Nothing
   it "handles the odd /- in gitlab URL" $ do
     let match = check [fmt|{domainName}/guibou/krank/-/issues/3|]
-    match `shouldBe` Just (GitIssue domain "guibou" "krank" 3)
+    match `shouldBe` Just (GitIssueRef domain "guibou" "krank" 3)
   it "handles long gitlab url with groups" $ do
     let match = check [fmt|{domainName}/gbataille_main/sub_level_1/sub_level_2/deep_in_groups/issues/3|]
-    match `shouldBe` Just (GitIssue domain "gbataille_main/sub_level_1/sub_level_2" "deep_in_groups" 3)
+    match `shouldBe` Just (GitIssueRef domain "gbataille_main/sub_level_1/sub_level_2" "deep_in_groups" 3)
   it "handles long gitlab url with groups and with the odd /-" $ do
     let match = check [fmt|{domainName}/gbataille_main/sub_level_1/sub_level_2/deep_in_groups/-/issues/12|]
-    match `shouldBe` Just (GitIssue domain "gbataille_main/sub_level_1/sub_level_2" "deep_in_groups" 12)
+    match `shouldBe` Just (GitIssueRef domain "gbataille_main/sub_level_1/sub_level_2" "deep_in_groups" 12)
 
 spec :: Spec
 spec = do
@@ -127,10 +127,10 @@ spec = do
         lalala https://gitlab.haskell.org/ghc/ghc/issues/16955
         |]
         match
-          `shouldMatchList` [ Localized (SourcePos "localFile" 1 1) $ GitIssue Github "guibou" "krank" 2,
-                              Localized (SourcePos "localFile" 3 9) $ GitIssue (Gitlab (GitlabHost "gitlab.com")) "gitlab-org" "gitlab-foss" 67390,
-                              Localized (SourcePos "localFile" 4 25) $ GitIssue Github "guibou" "krank" 1,
-                              Localized (SourcePos "localFile" 5 16) $ GitIssue (Gitlab (GitlabHost "gitlab.haskell.org")) "ghc" "ghc" 16955
+          `shouldMatchList` [ Localized (SourcePos "localFile" 1 1) $ GitIssueRef Github "guibou" "krank" 2,
+                              Localized (SourcePos "localFile" 3 9) $ GitIssueRef (Gitlab (GitlabHost "gitlab.com")) "gitlab-org" "gitlab-foss" 67390,
+                              Localized (SourcePos "localFile" 4 25) $ GitIssueRef Github "guibou" "krank" 1,
+                              Localized (SourcePos "localFile" 5 16) $ GitIssueRef (Gitlab (GitlabHost "gitlab.haskell.org")) "ghc" "ghc" 16955
                             ]
   describe "huge test" $ do
     let config =
@@ -219,14 +219,14 @@ spec = do
   describe "it parses when there is two url on the same line" $ do
     it "works correctly with only one in second position" $ do
       extractIssues "foo" "https://ip.tyk.nu https://github.com/x/x/issues/32"
-        `shouldBe` [ Localized (SourcePos "foo" 1 19) (GitIssue Github "x" "x" 32)
+        `shouldBe` [ Localized (SourcePos "foo" 1 19) (GitIssueRef Github "x" "x" 32)
                    ]
     it "works correctly with only one in first position" $ do
       extractIssues "foo" "foo bar baz https://github.com/x/x/issues/32 https://ip.tyk.nu"
-        `shouldBe` [ Localized (SourcePos "foo" 1 13) (GitIssue Github "x" "x" 32)
+        `shouldBe` [ Localized (SourcePos "foo" 1 13) (GitIssueRef Github "x" "x" 32)
                    ]
     it "works correctly with two correct url" $ do
       extractIssues "foo" "foo gitlab.com/foo/br/issues/10 https://github.com/x/x/issues/32 https://ip.tyk.nu"
-        `shouldBe` [ Localized (SourcePos "foo" 1 5) (GitIssue (Gitlab (GitlabHost "gitlab.com")) "foo" "br" 10),
-                     Localized (SourcePos "foo" 1 33) (GitIssue Github "x" "x" 32)
+        `shouldBe` [ Localized (SourcePos "foo" 1 5) (GitIssueRef (Gitlab (GitlabHost "gitlab.com")) "foo" "br" 10),
+                     Localized (SourcePos "foo" 1 33) (GitIssueRef Github "x" "x" 32)
                    ]

--- a/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
+++ b/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
@@ -190,14 +190,14 @@ spec = do
                      )
                    )
     it "ignore are ignored" $ do
-      let config =
+      let testConfig =
             KrankConfig
               { githubKey = Nothing,
                 gitlabKeys = Map.empty,
                 dryRun = False,
                 useColors = False
               }
-          env =
+          testEnv =
             TestEnv
               { envFiles = Map.singleton "foo" " hello you https://github.com/foo/bar/issues/10 yeah# krank:ignore-line\nhttps://github.com/foo/bar/issues/11",
                 envRestAnswers =
@@ -206,7 +206,7 @@ spec = do
                       (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "11", Right $ object [("state", String "open")])
                     ]
               }
-      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (env, config)
+      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (testEnv, testConfig)
       -- TODO: perhaps ignored lines must appears in the listing, but not as error
       res
         `shouldBe` ( False,

--- a/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
+++ b/tests/Test/Krank/Checkers/IssueTrackerSpec.hs
@@ -140,38 +140,42 @@ spec = do
               dryRun = False,
               useColors = False
             }
-        env state10 =
+        env state10 title10 title11 =
           TestEnv
             { envFiles = Map.singleton "foo" " hello you https://github.com/foo/bar/issues/10 yeah\nhttps://github.com/foo/bar/issues/11",
               envRestAnswers =
                 Map.fromList
-                  [ (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "10", Right $ object [("state", String state10)]),
-                    (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "11", Right $ object [("state", String "open")])
+                  [ (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "10", Right $ object [("state", String state10), ("title", String title10)]),
+                    (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "11", Right $ object [("state", String "open"), ("title", String title11)])
                   ]
             }
     it "should work" $ do
-      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (env "closed", config)
+      let firstIssueTitle = "fooobar"
+      let secondIssueTitle = "barbaz"
+      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (env "closed" firstIssueTitle secondIssueTitle, config)
       res
         `shouldBe` ( False,
-                     ( ["\nfoo:1:12: error:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    now Closed - You can remove the workaround you used there\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    still Open\n"] :: [Text],
-                       [ "Error when processing bar: user error (file not found)"
-                       ] ::
-                         [Text]
+                     ( [[fmt|\nfoo:1:12: error:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    the issue is now Closed - You can remove the workaround you used there\n    | title: {firstIssueTitle}\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    the issue is still Open\n    | title: {secondIssueTitle}\n|]] :: [Text],
+                       ["Error when processing bar: user error (file not found)"] :: [Text]
                      )
                    )
     it "with error in url" $ do
-      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo"])) (env "closed", config)
+      let firstIssueTitle = "barbouze"
+      let secondIssueTitle = "bouya"
+      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo"])) (env "closed" firstIssueTitle secondIssueTitle, config)
       res
         `shouldBe` ( False,
-                     ( ["\nfoo:1:12: error:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    now Closed - You can remove the workaround you used there\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    still Open\n"],
+                     ( [[fmt|\nfoo:1:12: error:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    the issue is now Closed - You can remove the workaround you used there\n    | title: {firstIssueTitle}\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    the issue is still Open\n    | title: {secondIssueTitle}\n|]],
                        [] :: [Text]
                      )
                    )
     it "with error in file" $ do
-      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (env "open", config)
+      let firstIssueTitle = "yazu"
+      let secondIssueTitle = "zuma"
+      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (env "open" firstIssueTitle secondIssueTitle, config)
       res
         `shouldBe` ( False,
-                     ( ["\nfoo:1:12: info:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    still Open\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    still Open\n"] :: [Text],
+                     ( [[fmt|\nfoo:1:12: info:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    the issue is still Open\n    | title: {firstIssueTitle}\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    the issue is still Open\n    | title: {secondIssueTitle}\n|]] :: [Text],
                        [ "Error when processing bar: user error (file not found)"
                        ] ::
                          [Text]
@@ -182,14 +186,18 @@ spec = do
     -- works correctly.
     -- Or we need to move more things inside the Krank monad, such as "exitFailure".
     it "without error" $ do
-      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo"])) (env "open", config)
+      let firstIssueTitle = "malaria"
+      let secondIssueTitle = "ria riu rio"
+      let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo"])) (env "open" firstIssueTitle secondIssueTitle, config)
       res
         `shouldBe` ( True,
-                     ( ["\nfoo:1:12: info:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    still Open\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    still Open\n"] :: [Text],
+                     ( [[fmt|\nfoo:1:12: info:\n  IssueTracker check for https://github.com/foo/bar/issues/10\n    the issue is still Open\n    | title: {firstIssueTitle}\n\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    the issue is still Open\n    | title: {secondIssueTitle}\n|]] :: [Text],
                        [] :: [Text]
                      )
                    )
     it "ignore are ignored" $ do
+      let firstIssueTitle = "rio del mare"
+      let secondIssueTitle = "margulin"
       let testConfig =
             KrankConfig
               { githubKey = Nothing,
@@ -202,15 +210,15 @@ spec = do
               { envFiles = Map.singleton "foo" " hello you https://github.com/foo/bar/issues/10 yeah# krank:ignore-line\nhttps://github.com/foo/bar/issues/11",
                 envRestAnswers =
                   Map.fromList
-                    [ (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "10", Right $ object [("state", String "closed")]),
-                      (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "11", Right $ object [("state", String "open")])
+                    [ (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "10", Right $ object [("state", String "closed"), ("title", String firstIssueTitle)]),
+                      (Req.https "api.github.com" Req./: "repos" Req./: "foo" Req./: "bar" Req./: "issues" Req./: "11", Right $ object [("state", String "open"), ("title", String secondIssueTitle)])
                     ]
               }
       let Right res = runReaderT (runWriterT (unTestKrank $ runKrank ["foo", "bar"])) (testEnv, testConfig)
       -- TODO: perhaps ignored lines must appears in the listing, but not as error
       res
         `shouldBe` ( False,
-                     ( ["\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    still Open\n"] :: [Text],
+                     ( [[fmt|\nfoo:2:1: info:\n  IssueTracker check for https://github.com/foo/bar/issues/11\n    the issue is still Open\n    | title: {secondIssueTitle}\n|]] :: [Text],
                        [ "Error when processing bar: user error (file not found)"
                        ] ::
                          [Text]

